src/transport/UnauthenticatedSessionTable.h

```diff
@@ -132,6 +134,23 @@ class UnauthenticatedSession : public Session,
 
     PeerMessageCounter & GetPeerMessageCounter() { return mPeerMessageCounter; }
 
+    static void Release(UnauthenticatedSession * obj)
```

Comment:
Tentatively wait for approval from @kpschoedel

Reply:
Looks fine. Not thrilled by having more `#if`s, but I can't think of anything better here.